### ISSUE TITLE
Fork to template.github.project

### ISSUE THREAD

**Issue by devlinjunker** [devlinjunker/template.github#2]
**Wait for #6, #8 (and maybe #4) to complete before forking**
- [ ] more docs
    - initial (and other) design/architecture docs (see #7)
    - deployment
    - release process
    - (Test/API First?) steps to create a feature
       1. Design
          - Create Design Document (#7)
          - Create Mockups (HTML/CSS)?
          - **Optional: Define OpenAPI YAML file for API** - If designed by non-technical
       2. Backend
          - Expected Parameters
          - Example Responses
       3. API
          - Define data models (and validation?)
              - examples in `.json` files
          - Define Endpoint Unit Tests and expect to fail at first
              - "Should use..."
              - "Should error ..", 
              - "Should set/change .. in db", 
              - "Should cause...", 
              - "Should return...." (is this Newman?)
          - Create stubs with request/response definitions
          - **Generate OpenAPI YAML file of API** 
             - Autogenerate Newman Tests
             - Compare with Design OpenAPI YAML File -- **If Exists**
          - Build to meet Unit + Newman Tests
          - First version release with a single working endpoint is 0.1.0
             - increment minor until ready for release, then increment major
       4. UI
          - Autogenerate Mock Server for UI Development **with OpenAPI YAML**
          - Autogenerate js library **with OpenAPI YAML**
          - Build HTML/CSS Components from Mockup and Style Guide
          - Create Unit tests for components
             - "Should fetch data from ..."
             - "Should .. on error/bad data"
             - "Should change state with ..."
            -  "Should send message to ..."
            -  "Should show ..." (unsure?)
           - Create Integration Tests with Cucumber and Protractor?
           - Create JS Components that meet Unit Tests that use HTML/CSS
           - Incorporate and Test Integration Tests
  
- [ ] projects
    - Parking Lot? (Where extremely low priority things go -- assigned `wontfix` label?) or just remove from backlog
    - Backlog
       - New Column 
       - Blocked column
       - Priority column
    - Sprint? 
       - Open
       - In Progress
       - Under Review
       - Merged

- [ ] advanced github actions:
    - https://github.com/technote-space/create-project-card-action
        - creates card in backlog when issue added
    - https://github.com/technote-space/auto-card-labeler
        - updates label based on columns in projects (sprint and backlog)
        - vice versa? Update column based on label?
    - add good first issue label to score: 1 and 2
    - Prevent `status: blocked` labeled commits or branches from being merged
        - more complicated... need a way to assign `blocked` label from other issues/prs
        - if comment contains “blocked by” or “blocks” set blocked label?
        - unblock by removing label

- [ ] Dashboard pages
   - current PRs open
   - “new” (not responded to and/or within 1-3 months)
   - issue sorting/filtering based on labels/projects/milestones
   - priority items
   - todo search 
   - bug tracker (by week/release)
   - backlog page

- [ ] Look into milestones
   - `future` label for things with no release date
   - dashboard with (-2) prev, current, upcoming(+4) and future milestones 

- [ ] pull request process
   - draft PR for each FULL feature branch
      - contain links to design/todos(that can become separate PRs)
   - can filter these out of searches until they are ready `!is:draft`
   - subtask `feature/ ` branches/PRs merged to FULL feature branch
      - recommended when FULL feature branch gets lots of changes/commits
         - create new branch
         - reset FULL feature to first commit
         - create PR into FULL feature from new branch
      - see #23 for limits/recommendation to split PR
      - action that checks out and runs script to see how large PR is?
         - can block merges unless forced by admin

- [ ] healthcheck checker that sees if services are up
   - config.yaml file that manages the services
   - html page with js that calls all these services and compares responses
       - e.g https://github.com/devlinjunker/template.webpack.fend/issues/33 or https://github.com/devlinjunker/template.hapi.rest/pull/2
       - generalize so expected response can be configured?
          - or maybe come up with standard health check look that all must follow
   - hosted on github.io?
   - should this be an "overarching" project that is repo containing links to multiple repos?

- [ ] labels
   - remove effort labels from template.github and only include in Project template
   - `backlog` label

**Comment by devlinjunker** [comment]
Should complexity labels be moved to this project and removed from template.github?

**Comment by devlinjunker** [comment]
(From #7)
- [ ] Design doc:
  - what problem you are trying to solve
  - who is this being built for
  - background/domain knowledge/terminology
  - use cases
  - requirements
  - existing solutions (architecture/design decisions)
  - stakeholders

**Comment by devlinjunker** [comment]
blocked by:
- [ ] #55 
- [ ] #53

**Comment by devlinjunker** [comment]
Extension:
 - shows warning if backlog has too many items

**Comment by devlinjunker** [comment]
Waiting on "template.npm.semver" project and "js.api" repo (with github api) for quick dashboard creation?
